lib/ansible/modules/windows/win_xml.ps1

```diff
@@ -0,0 +1,182 @@
+                throw "attribute value mismatch for actual=" + $actual.Name
+            }
+        }
+    }
+
+    ##children
+    if ($expected.ChildNodes.Count -ne $actual.ChildNodes.Count)  {
+        throw "child node mismatch. for actual=" + $actual.Name
+    }
+
+    for ($i=0;$i -lt $expected.ChildNodes.Count; $i =$i+1) {
+        if (-not $actual.ChildNodes[$i]) {
+            throw "actual missing child nodes. for actual=" + $actual.Name
+        }
+        Compare-XmlDocs $expected.ChildNodes[$i] $actual.ChildNodes[$i]
+    }
+
+    if ($expected.InnerText) {
+        if ($expected.InnerText -ne $actual.InnerText) {
+           throw "inner text mismatch for actual=" + $actual.Name
+        }
+    }
+    elseif ($actual.InnerText) {
+        throw "actual has inner text but expected does not for actual=" + $actual.Name
+    }
+}
+
+function BackupFile($path) {
+	$backuppath = $path + "." + [DateTime]::Now.ToString("yyyyMMdd-HHmmss");
+	Copy-Item $path $backuppath;
+	return $backuppath;
+}
+
+$params = Parse-Args $args -supports_check_mode $true
+$check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -type "bool" -default $false
+
+$dest = Get-AnsibleParam $params "path" -type "path" -FailIfEmpty $true -aliases "dest", "file"
+$fragment = Get-AnsibleParam $params "fragment" -type "str" -FailIfEmpty $true -aliases "xmlstring"
+$xpath = Get-AnsibleParam $params "xpath" -type "str" -FailIfEmpty $true
+$backup = Get-AnsibleParam $params "backup" -type "bool" -Default "no" -ValidateSet "no", "yes"
+$type = Get-AnsibleParam $params "type" -type "str" -Default "element" -ValidateSet "element", "attribute", "text"
+$attribute = Get-AnsibleParam $params "attribute" -type "str" -FailIfEmpty ($type -eq "attribute")
+
+$result = @{
+    changed = $false
+}
+
+If (-Not (Test-Path -Path $dest -PathType Leaf)){
+    Fail-Json $result "Specified path $dest does not exist or is not a file."
+}
+
+[xml]$xmlorig = Get-Content -Path $dest
+$xmlnew = $xmlorig.Clone()
+if ($type -eq "element") {
+    $xmlchild = [xml]$fragment
+
+    $child = $xmlnew.CreateElement($xmlchild.DocumentElement.Name, $xmlnew.DocumentElement.NamespaceURI)
+    foreach ($attr in $xmlchild.DocumentElement.Attributes) {
+        $child.SetAttribute($attr);
+    }
+
+    foreach ($node in $xmlchild.DocumentElement.ChildNodes) {
+        $newnode = $xmlnew.CreateElement($node.Name, $xmlnew.DocumentElement.NamespaceURI)
+        foreach ($attr in $node.Attributes) {
+            $newnode.SetAttribute($attr)
+        }
+        $newnode.InnerText = $node.InnerText
+        $child.AppendChild($newnode) | Out-Null
+    }
+
+    $node = $xmlnew.SelectSingleNode($xpath)
+    $elements = $node.ChildNodes
+    [bool]$add = $TRUE
+    if ($elements.Count) {
+        foreach ($element in $elements) {
+            try {
+                Compare-XmlDocs $xmlchild.DocumentElement $element
+                $add = $FALSE
+                break
```

Comment:
I might be missing something here, why are we breaking the loop after checking the first element, shouldn't `Compare-XmlDocs` be run on all the elements?